src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/RestoreTelemetryEvent.cs

```diff
@@ -28,6 +28,15 @@ public class RestoreTelemetryEvent : ActionEventBase
             int packageCount,
             int noOpProjectsCount,
             int upToDateProjectsCount,
+            int unknownProjectsCount,
```

Comment:
many ints in a row can lead to a bad telemetry (as I learned in https://github.com/NuGet/NuGet.Client/commit/14fa722fbfeb9b485e86dac5c20e589de9138bf5#diff-3d7ee2d46330653387ac5c5e3aea0721)

Double/triple/quadruple check the numbers are right.

Reply:
I am thinking of changing the constructor to accept an object of type `IEnumerable<NuGetProject>` and make the list calculations inside the constructor. That way, we don't need to worry about the integers

Reply:
I think it's better to keep current approach with the counters in constructor

Reply:
Thinking again. I think it's better to use `IEnumerable<NuGetProject>` because:

- This will only affect `RestoreTelemetryEvent` class, not the base class that uses project ID's string array
- Avoids possible assignation issues
- Counters are a 'calculated property' from the projects list
- Either way, I'm changing the constructor

Reply:
My `IEnumerable<NuGetProject>` approach will not work because of base constructor.

Reply:
I just double checked the counters assignations and those are OK.